langtest/transform/robustness.py

```diff
@@ -503,14 +505,27 @@ def transform(
                             break
 
                 replace_token = sent_tokens[replace_idx : replace_idx + len(replace_idxs)]
+                token_length = len(replace_token)
                 replace_token = " ".join(replace_token)
-
-                chosen_ent = random.choice(terminology[ent_type])
+                filtered_ents = [
+                    ent
+                    for ent in terminology[ent_type]
+                    if len(ent.split(" ")) == token_length
+                ]
+                if not filtered_ents:
+                    chosen_ent = replace_token
+                else:
+                    chosen_ent = random.choice(filtered_ents)
```

Comment:
Not sure to get what's cooking here..

Reply:
We look if we have same length replacement (that are normally there) If not there, we choose randomly. However while doing it randomly, we found that we were getting absurd replacements.